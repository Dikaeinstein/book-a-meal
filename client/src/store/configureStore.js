import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
