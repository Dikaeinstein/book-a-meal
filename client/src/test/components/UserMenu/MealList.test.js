import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedMealList, { MealList } from '../../../components/UserMenu/MealList';

describe('<MealList />', () => {
  const meals = [
    { id: 1, name: 'test meal1' },
    { id: 2, name: 'test meal2' },
    { id: 3, name: 'test meal3' },
  ];
  it('should render with correct props', () => {
    const mockStore = configureStore();
    const store = mockStore({
      menu: {
        byId: {
          1: { name: 'test menu', meals },
        },
        menuId: 1,
      },
    });

    const wrapper = shallow(<ConnectedMealList store={store} link="/test" />);
    expect(wrapper.props().meals).toEqual(meals);
    expect(wrapper.props().link).toEqual('/test');
  });
  it('should match snapshot', () => {
    const tree = shallow(<MealList meals={meals} link="/" />);
    expect(tree).toMatchSnapshot();
  });
});
