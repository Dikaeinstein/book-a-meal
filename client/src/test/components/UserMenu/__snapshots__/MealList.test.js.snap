// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MealList /> should match snapshot 1`] = `
<section
  className="card user-menu text-left"
>
  <Connect(MenuMeal)
    defaultMeal={
      Object {
        "id": 1,
        "name": "test meal1",
      }
    }
    key="1"
    link="/"
  />
  <Connect(MenuMeal)
    defaultMeal={
      Object {
        "id": 2,
        "name": "test meal2",
      }
    }
    key="2"
    link="/"
  />
  <Connect(MenuMeal)
    defaultMeal={
      Object {
        "id": 3,
        "name": "test meal3",
      }
    }
    key="3"
    link="/"
  />
</section>
`;
